File: src/main/java/com/leijendary/spring/microservicetemplate/document/SampleTranslationDocument.java
package com.leijendary.spring.microservicetemplate.document;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.elasticsearch.annotations.Field;

import static org.springframework.data.elasticsearch.annotations.FieldType.Text;

@Data
@EqualsAndHashCode(callSuper = true)
public class SampleTranslationDocument extends LocaleDocument {

    @Field(type = Text)
    private String name;

    @Field(type = Text)
    private String description;
}

File: src/main/kotlin/com/leijendary/spring/template/document/SampleTranslationDocument.kt
package com.leijendary.spring.template.document

import com.leijendary.spring.template.core.document.LocaleDocument
import org.springframework.data.elasticsearch.annotations.Field
import org.springframework.data.elasticsearch.annotations.FieldType.Text

class SampleTranslationDocument(
    @Field(type = Text, analyzer = "ngram_analyzer", searchAnalyzer = "standard")
    var name: String,

    @Field(type = Text, analyzer = "ngram_analyzer", searchAnalyzer = "standard")
    var description: String
) : LocaleDocument()